generators/language-java/templates/java-spring/src/main/java/application/ibmcloud/CloudServices.java

```diff
@@ -141,8 +144,53 @@ private String getJsonValue(String jsonPath, String json) {
         }
         return value;
     }
-    
+  
     // Search pattern resolvers
+	
+    private String getUserProvidedValue(String pattern) {
+        LOGGER.info("user-provided entry found:  " + pattern);
+        String value = null;
+        String vcap_services = System.getenv(VCAP_SERVICES);
+        if (vcap_services == null || vcap_services.isEmpty() || pattern == null) {
+            LOGGER.info("No VCAP_SERVICES or no user-provided pattern");
+            return null;
+        }
+        int i = pattern.indexOf(":");
```

Comment:
maybe use `lastIndexOf`. I dont think the name is supposed to have `:`'s in it, but hey, defensiveness never hurt